WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Package and Rate Selection/WooShippingAddPackageView.swift

```diff
@@ -0,0 +1,254 @@
+    var addPackageButtonDisabled: Bool {
+        for (_, value) in fieldValues {
+            if value.isEmpty {
+                return true
+            }
+        }
+        return fieldValues.count != WooShippingAddPackageDimensionView.DimensionType.allCases.count
+    }
+
+    var body: some View {
+        NavigationView {
+            VStack {
+                Picker("", selection: $selectedPackageType) {
+                    ForEach(PackageProviderType.allCases, id: \.self) {
+                        Text($0.name)
+                    }
+                }
+                .pickerStyle(.segmented)
+                .padding()
+                Spacer()
+                selectedPackageTypeView
+                Spacer()
+                Button(Localization.addPackage) {
+                    // add package
+                }
+                .buttonStyle(PrimaryButtonStyle())
+                .disabled(addPackageButtonDisabled)
+                .padding()
+            }
+            .toolbar {
+                ToolbarItem(placement: .topBarLeading) {
+                    Button(action: {
+                        presentationMode.wrappedValue.dismiss()
+                    }, label: {
+                        Text(Localization.cancel)
+                    })
+                }
+            }
+            .navigationTitle(Localization.addPackage)
+            .navigationBarTitleDisplayMode(.inline)
+        }
+        .navigationViewStyle(.stack)
+    }
+
+    @ViewBuilder
+    private var selectedPackageTypeView: some View {
+        ScrollView {
+            switch selectedPackageType {
+            case .custom:
+                customPackageView
+            case .carrier:
+                carrierPackageView
+            case .saved:
+                savedPackageView
+            }
+        }
+    }
+
+    enum PackageType: CaseIterable {
+        case box, envelope
+        var name: String {
+            switch self {
+            case .box:
+                return Localization.box
+            case .envelope:
+                return Localization.envelope
+            }
+        }
+    }
+
+    @State var packageType: PackageType = .box
+    @State var showSaveTemplate: Bool = false
+    @State var length: String = ""
+    @State var fieldValues: [WooShippingAddPackageDimensionView.DimensionType: String] = [:]
+
+    private var customPackageView: some View {
+        VStack(alignment: .leading) {
+            HStack {
+                Text(Localization.packageType)
+                    .font(.subheadline)
+                Spacer()
+            }
+            // type selection
+            Menu {
+                // show selection
+                ForEach(PackageType.allCases, id: \.self) { option in
+                    Button {
+                        packageType = option
+                    } label: {
+                        Text(option.name)
+                            .font(.body)
```

Comment:
Just a small suggested addition here, to show a checkmark next to the selected package type:

```suggestion
                        Text(option.name)
                            .font(.body)
                        if packageType == option {
                            Image(uiImage: .checkmarkStyledImage)
                        }
```

Reply:
Good point! Missed that one in figma!